include/ifc/pathname.hxx

```diff
@@ -103,8 +104,9 @@ namespace ifc {
 
         value_type back() const
         {
-            IFCASSERT(length() > 0);
-            return begin()[length() - 1];
+            std::size_t index = length();
+            IFCASSERT(index != 0);
+            return begin()[static_cast<difference_type>(index - 1)];
         }
```

Comment:
Why the static_cast here to `difference_type`?  Is that what GCC + Clang want? 

In general, we should add cast only where necessary or clarify intent.

Reply:
`iterator::operator[]` takes `difference_type`, which is `std::ptrdiff_t`, but we pass `std::size_t` here. This results in a sign mismatch on all platforms.

Reply:
Ah, OK.  I forgot about that.  Sorry!
I suppose we would just have to rewrite this class, but not in this PR!

Every time we have to use a cast, that indicates something fishy in the design :-)